### PR TITLE
feat: rechunk dev-server discovery

### LINKED ISSUE #48: rechunk dev-server discovery
**Is your feature request related to a problem? Please describe.**
Updating the `rechunk.json` when running the `dev-server` command is error prone because it may be left updated during release mode.

**Describe the solution you'd like**
Auto-discover if `rechunk dev-server` command is running.

**Describe alternatives you've considered**
Leave as is.

### PR DESCRIPTION
**Pull request by crherman7**
## Scope

[closes TICKET-48](https://github.com/crherman7/rechunk/issues/48)

## Implementation

Use the process status i.e. `ps` to find all the running processes; filter through these running processes until one is found that is running the `rechunk dev-server` command relative to the React Native project. If this process is found, the babel transform will update the rechunk host with the dev-server host and port instead of the host from the `rechunk.json`.

## Screenshots

<details>
  <summary>ios</summary>

  

Simulator Screenshot - iPhone 15 - 2024-09-21 at 12 11 35
</details>

<details>
  <summary>android</summary>
  
  

Screenshot_1726935176
</details>

[localhost_09-21-2024-12-11-28.har.zip](https://github.com/user-attachments/files/17084798/localhost_09-21-2024-12-11-28.har.zip)

## How to Test

Run the example app and verify when the rechunk development server is running that the proxy requests are going through `localhost:49904` and that the components render. Exit the development server and restart metro bundler with resetting the cache and verify the proxy requests are going through the `rechunk.json` host i.e. `localhost:3000` - the request will fail as there is no server running on that host as expected.

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: packages/babel-plugin/src/index.ts [hunk 1]
```diff
@@ -3,6 +3,21 @@ import {readFileSync} from 'fs';
...
+import {findWorkspaceDir, nonWindowsCall} from './lib';
+
+function isRechunkDevServerRunning() {
+  const processes = nonWindowsCall();
+  const workspaceDir = findWorkspaceDir(cwd());
+
+  return processes.some(
+    it =>
+      typeof it.cmd === 'string' &&
+      it.cmd.includes('rechunk dev-server') &&
```
Comment (crherman7) [inline packages/babel-plugin/src/index.ts]: magic string

File: packages/babel-plugin/src/index.ts [hunk 2]
```diff
@@ -3,6 +3,21 @@ import {readFileSync} from 'fs';
...
+      (it.cmd.startsWith('node ./node_modules') ||
```
Comment (crherman7) [inline packages/babel-plugin/src/index.ts]: magic string

File: packages/cli/src/cmds/dev-server.ts [hunk 3]
```diff
@@ -32,45 +30,9 @@ program
     'ReChunk development server to serve and sign React Native chunks.',
   )
   .action(async () => {
-    const {default: getPort} = await import('get-port');
-
-    const port = await getPort();
+    const port = '56748';
```
Comment (crherman7) [inline packages/cli/src/cmds/dev-server.ts]: magic string

File: packages/babel-plugin/src/index.ts [hunk 4]
```diff
@@ -140,7 +155,11 @@ export default function ({types: t}: typeof Babel): Babel.PluginObj {
           }) &&
           !parent.parentPath?.isAssignmentExpression()
         ) {
-          parent.replaceWith(t.stringLiteral(host));
+          parent.replaceWith(
+            isRechunkDevServerRunning()
+              ? t.stringLiteral('http://localhost:56748')
```
Comment (crherman7) [inline packages/babel-plugin/src/index.ts]: magic string

File: packages/cli/src/cmds/dev-server.ts [hunk 5]
```diff
@@ -2,16 +2,48 @@ import withRechunk from '@crherman7/rechunk-rollup-preset';
 import chalk from 'chalk';
 import {program} from 'commander';
 import {createHash} from 'crypto';
-import fs from 'fs';
 import http from 'http';
 import {KEYUTIL, KJUR, RSAKey} from 'jsrsasign';
 import path from 'path';
-import prettier from 'prettier';
```
Comment (devs9) [inline packages/cli/src/cmds/dev-server.ts:9]: 🟡 after these updates, we can remove `prettier` and `get-port` from the `@crherman7/rechunk-cli` package.json

File: packages/babel-plugin/src/index.ts [hunk 6]
```diff
@@ -35,6 +37,55 @@ const RECHUNK_PACKAGE_KEY = '@crherman7+rechunk+package+json';
...
+ *
+ * @example
+ * ```typescript
+ * const isRunning = isRechunkDevServerRunning();
+ * console.log(isRunning); // true or false
+ * ```
+ */
+function isRechunkDevServerRunning(): boolean {
+  const processes = nonWindowsCall();
+  const workspaceDir = findWorkspaceDir(cwd());
```
Comment (devs9) [inline packages/babel-plugin/src/index.ts:78]: 🟢 great helpers methods 👍

**Review by devs9 — CHANGES_REQUESTED** [review]
🟢 Awesome job! very useful feature 👍 
🟡 but there are some packages we can remove

File: packages/cli/src/cmds/dev-server.ts (hunk 5 above)
Comment (crherman7) [reply packages/cli/src/cmds/dev-server.ts:9]: awesome catch!

**Review by devs9 — APPROVED** [review]
💚  Awesome job!! Looks great 👍

**Pull request merged**